test/box-tap/check_merge_source.c

```diff
@@ -72,7 +74,7 @@ lbox_check_merge_source_call_next(struct lua_State *L)
 	 * implementation detail and the test looks more clean
 	 * when we don't lean on this fact.
 	 */
-	struct lua_State *temporary_L = fiber_self()->storage.lua.stack;
+	struct lua_State *temporary_L = fiber_lua_state_internal(fiber_self());
```

Comment:
Why can't you access struct fiber? You have fiber.h available here. Just include it. I don't understand why do you need to change anything in this file. It can use our headers and even link with any of our libraries.

Reply:
Because in this case the `enum say_level` will be declared twice (from `module.h` and from `fiber.h -> diag.h -> say.h`).

Reply:
I had a conversation with Vlad and it looks like he won the fight. We can include error.h instead of module.h and not export any extra symbols (I mean diag_get or fiber_lua_state_internal).
Updated.